draft-ietf-tls-tls13.md

```diff
@@ -2323,102 +2315,138 @@ The "extension_data" field of this extension contains a
 
 %%% Signature Algorithm Extension
        enum {
-           none(0),
-           md5_RESERVED(1),
-           sha1(2),
-           sha224_RESERVED(3),
-           sha256(4), sha384(5), sha512(6),
-           (255)
-       } HashAlgorithm;
+           // RSASSA-PKCS-v1_5 algorithms.
+           rsa_pkcs1_sha1 (0x0201),
+           rsa_pkcs1_sha256 (0x0401),
+           rsa_pkcs1_sha384 (0x0501),
+           rsa_pkcs1_sha512 (0x0601),
+
+           // DSA algorithms (deprecated).
+           dsa_sha1 (0x0202),
+           dsa_sha256 (0x0402),
+           dsa_sha384 (0x0502),
+           dsa_sha512 (0x0602),
+
+           // ECDSA algorithms.
+           ecdsa_secp256r1_sha256 (0x0403),
+           ecdsa_secp384r1_sha384 (0x0503),
+           ecdsa_secp521r1_sha512 (0x0603),
+
+           // RSASSA-PSS algorithms.
+           rsa_pss_sha256 (0x0700),
+           rsa_pss_sha384 (0x0701),
+           rsa_pss_sha512 (0x0702),
+
+           // EdDSA algorithms.
+           ed25519 (0x0703),
+           ed448 (0x0704),
 
-       enum {
-           anonymous_RESERVED(0),
-           rsa(1),
-           dsa(2),
-           ecdsa(3),
-           rsapss(4),
-           eddsa(5),
-           (255)
+           // Reserved Code Points.
+           obsolete_RESERVED (0x0000..0x0200),
+           obsolete_RESERVED (0x0203..0x0400),
+           obsolete_RESERVED (0x0404..0x0500),
+           obsolete_RESERVED (0x0504..0x0600),
+           obsolete_RESERVED (0x0604..0x06FF),
+           private_use (0xFE00..0xFFFF),
+           (0xFFFF)
        } SignatureAlgorithm;
 
-       struct {
-           HashAlgorithm hash;
-           SignatureAlgorithm signature;
-       } SignatureAndHashAlgorithm;
-
-       SignatureAndHashAlgorithm
-         supported_signature_algorithms<2..2^16-2>;
-
-[[TODO: IANA considerations for new SignatureAlgorithm value]]
-
-Each SignatureAndHashAlgorithm value lists a single hash/signature pair that
-the client is willing to verify. The values are indicated in descending order
-of preference.
-
-Note: Because not all signature algorithms and hash algorithms may be accepted
-by an implementation (e.g., ECDSA with SHA-256, but not SHA-384), algorithms here
-are listed in pairs.
-
-hash
-: This field indicates the hash algorithms which may be used.  The
-  values indicate support for unhashed data, SHA-1, SHA-256, SHA-384,
-  and SHA-512 {{SHS}}, respectively. The "none" value is provided for
-  signature algorithms which do not require hashing before signing,
-  such as EdDSA.
-
-  Previous versions of TLS
-  supported MD5, SHA-1, and SHA-224.  These algorithms are now deprecated.
-  MD5 and SHA-224 MUST NOT be offered by TLS 1.3 implementations; SHA-1 SHOULD
-  NOT be offered.
```

Comment:
That's in the paragraph in line 2430. Since we're no longer maintaining continuity with TLS 1.2's registry, the things that are MUST NOT (rather than SHOULD NOT) seemed simpler to just not define at all. This was done per mailing list feedback here:
http://www.ietf.org/mail-archive/web/tls/current/msg19070.html

(Although it cited MD5 and DSA. DSA is still mentioned because the existing text gave it a SHOULD NOT rather than MUST NOT.)